### PR TITLE
Fix plastic flaps walk over bug + PlasticFlapsAirtightOpaque

### PR DESCRIPTION
**Pull request by Cojoke-dot**
## About the PR

Prevent players from getting on tables and walking through plastic flaps + makes PlasticFlapsAirtightOpaque Opaque
## Why / Balance

#28877 and #22925 and #9660
## Technical details

adds HighImpassable to plastic flaps, child PlasticFlapsAirtightOpaque to PlasticFlapsAirtight, makes PlasticFlapsAirtightOpaque opaque and sets its required collisions. Creates a new component named ThrownItemAvoidComponent which indicates if something should have collisions ignored by thrown objects.
## Media

https://github.com/space-wizards/space-station-14/assets/83733158/866b2086-b2de-4225-8f10-4955707cbf8f
- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

NA
**Changelog**

:cl:
- fix: You can no longer stand on tables to walk through plastic flaps

### CODE REVIEW

**Comment by Cojoke-dot** [comment]
dam, I broke something

File: Resources/Prototypes/Entities/Structures/plastic_flaps.yml [hunk 1]
```diff
@@ -97,19 +99,25 @@
 
 - type: entity
   id: PlasticFlapsAirtightOpaque
-  parent: PlasticFlapsBase
+  parent: PlasticFlapsAirtightClear
```
Comment (ShadowCommander) [inline Resources/Prototypes/Entities/Structures/plastic_flaps.yml:102]: This should be PlasticFlapsOpaque
```suggestion
  parent: PlasticFlapsOpaque
```

File: Resources/Prototypes/Entities/Structures/plastic_flaps.yml [hunk 2]
```diff
@@ -97,19 +99,25 @@
...
+          !type:PhysShapeAabb
+          bounds: "-0.49,-0.49,0.49,0.49"
+        density: 100
+        mask:
+        - Impassable
+        layer:
+        - Opaque
+        - MidImpassable
+        - HighImpassable
+  - type: Occluder
```
Comment (ShadowCommander) [inline Resources/Prototypes/Entities/Structures/plastic_flaps.yml:120]: Don't need this if you inherit from PlasticFlapsOpaque

File: Resources/Prototypes/Entities/Structures/plastic_flaps.yml [hunk 3]
```diff
@@ -97,19 +99,25 @@
...
+  - type: Construction
+    graph: PlasticFlapsGraph
+    node: opaqueFlaps
```
Comment (ShadowCommander) [inline Resources/Prototypes/Entities/Structures/plastic_flaps.yml:123]: This should be a different node. It also probably needs a construction graph node in the construction YAML.

**Review by ShadowCommander — CHANGES_REQUESTED** [review]

File: Resources/Prototypes/Entities/Structures/plastic_flaps.yml (hunk 1 above)
Comment (Cojoke-dot) [reply Resources/Prototypes/Entities/Structures/plastic_flaps.yml:102]: PlasticFlapsOpaque has a construction graph, how would I disable the graph in the child? I attempted to do it but I only came up with errors.

File: Content.Shared/Throwing/ThrownItemSystem.cs [hunk 4]
```diff
@@ -70,7 +70,7 @@ private void HandleCollision(EntityUid uid, ThrownItemComponent component, ref S
 
         private void PreventCollision(EntityUid uid, ThrownItemComponent component, ref PreventCollideEvent args)
         {
-            if (args.OtherEntity == component.Thrower)
+            if (args.OtherEntity == component.Thrower || HasComp<ThrownItemAvoidComponent>(args.OtherEntity))
             {
                 args.Cancelled = true;
             }
```
Comment (metalgearsloth) [inline Content.Shared/Throwing/ThrownItemSystem.cs:76]: PreventCollideEvent is for 1 entity to not collide with another, not to ignore an entire class of entities. Use collision layers for this.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

**Pull request closed without merging**